File: Python/80. Remove Duplicates from Sorted Array II.py
from typing import List


# noinspection PyMethodMayBeStatic, PyPep8Naming
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        length = len(nums)

        if length < 3:
            return length

        left = 2

        for right in range(2, length):
            if nums[left - 2] != nums[right]:
                nums[left] = nums[right]
                left += 1

        return left

File: PHP/80. Remove Duplicates from Sorted Array II.php
<?php

/**
 *  @noinspection PhpIllegalPsrClassPathInspection, PhpMultipleClassDeclarationsInspection
 */
class Solution
{
    /**
     * @param  int[]  $nums
     *
     * @noinspection PhpUnused
     */
    public function removeDuplicates(array &$nums): int
    {
        $length = count($nums);

        if ($length < 3) {
            return $length;
        }

        $left = 2;

        for ($right = 2; $right < $length; $right++) {
            if ($nums[$left - 2] !== $nums[$right]) {
                $nums[$left++] = $nums[$right];
            }
        }

        return $left;
    }
}
